Text Generation WebUI/src/cert-files.ts

```diff
@@ -48,30 +50,60 @@ export const updateCertFilesIfNeeded = async (logger: Logger): Promise<void> =>
   }
   logger.debug('Cert files found in input data folder');
 
+  const input = await readCertFiles(certFilesInInputs);
+  const isInputValid = isCertValid({
+    certificate: input.fullchainPem,
+    privateKey: input.privateKeyPem,
+  });
+  if (!isInputValid) {
+    logger.debug('Input cert files are not valid');
+    return;
+  }
+
   const certFilesInSecrets = await findCertFiles(config.secretsDataFolder);
   if (!certFilesInSecrets) {
     logger.debug('No cert files found in secrets data folder');
   } else {
     logger.debug('Cert files found in secrets data folder');
   }
 
-  const needToValidateCertFiles = Boolean(certFilesInSecrets);
-  if (needToValidateCertFiles) {
-    // TODO: check if cert files are valid
-    logger.debug('Cert files validated');
-  }
+  const needToCopy = async (): Promise<boolean> => {
+    const isInputChainValid = verifyCertificateChain(parseFullchainPem(input.fullchainPem));
+    if (!isInputChainValid) {
+      logger.warn('Input certificate chain is not valid');
+      return false;
+    }
+
+    logger.debug('Input certificate chain is valid');
+
+    if (!certFilesInSecrets) {
+      return true;
+    }
+
+    const secret = await readCertFiles(certFilesInSecrets);
+    const secretCert = new X509Certificate(secret.fullchainPem);
+    const inputCert = new X509Certificate(input.fullchainPem);
 
-  await fs.copyFile(
-    certFilesInInputs.certFilePath,
-    path.resolve(config.secretsDataFolder, config.certFileName),
-  );
-  await fs.copyFile(
-    certFilesInInputs.certPrivateKeyPath,
-    path.resolve(config.secretsDataFolder, config.certPrivateKeyFileName),
-  );
-
-  logger.debug(
-    { certFilesInInputs, certFilesInSecrets },
-    'Copied cert files from input data folder to secrets data folder',
-  );
+    return new Date(inputCert.validTo) > new Date(secretCert.validTo);
```

Comment:
Also let check CN. It should be same in the secrets and in the inputs

Reply:
Done - https://github.com/Super-Protocol/solutions/pull/61/commits/f4d1ad9f3f8a7840112ae664ab60f0bb4d4349a1